ndk/src/main/jni/utils/string.c

```diff
@@ -0,0 +1,43 @@
+#include "string.h"
+#include <stdbool.h>
+#include <limits.h>
+#include <string.h>
+
+void bsg_strcpy(char *dst, char *src) {
+    bsg_strncpy(dst, src, INT_MAX);
+}
+
+void bsg_strncpy(char *dst, char *src, size_t len) {
+    int i = 0;
+    while (i <= len) {
+        char current = src[i];
+        dst[i] = current;
+        if (current == '\0') {
+            break;
+        }
+        i++;
+    }
+}
+
+size_t bsg_strlen(char *str) {
+  size_t i = 0;
+  while (true) {
+    char current = str[i];
+    if (current == '\0') {
+      return i;
+    } else if (i == INT_MAX) {
+      return INT_MAX;
+    }
+    i++;
+  }
+}
+
+void bsg_strncpy_safe(char *dst, char *src, int dst_size) {
```

Comment:
It would have to be something that calls this method, rather than here. Are there any additional details? I don't see any warnings during compilation and will turn on warnings as errors.

Reply:
I opened this file in Android Studio, and observed the warning `Parameter type mismatch: using size_t for signed values of typed int`.

Reply:
There has to be more info somewhere (maybe the Gradle console?) about what is calling this function with unexpected arguments. Warnings as errors is on in the NDK lib build now so anything incorrect, like implicit type coercion, breaks the build.